google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/SingleUseTransaction.java

```diff
@@ -582,6 +588,29 @@ private ApiFuture<Tuple<Long, ResultSet>> executeTransactionalUpdateAsync(
     return transactionalResult;
   }
 
+  private static final QueryUpdateOption[] LAST_STATEMENT_OPTIONS =
+      new QueryUpdateOption[] {Options.lastStatement()};
+
+  private static UpdateOption[] appendLastStatement(UpdateOption[] options) {
+    if (options.length == 0) {
+      return LAST_STATEMENT_OPTIONS;
+    }
+    UpdateOption[] result = new UpdateOption[options.length + 1];
+    System.arraycopy(options, 0, result, 0, options.length);
+    result[result.length - 1] = LAST_STATEMENT_OPTIONS[0];
+    return result;
+  }
+
+  private static QueryOption[] appendLastStatement(QueryOption[] options) {
+    if (options.length == 0) {
+      return LAST_STATEMENT_OPTIONS;
+    }
+    QueryOption[] result = new QueryOption[options.length + 1];
+    System.arraycopy(options, 0, result, 0, options.length);
+    result[result.length - 1] = LAST_STATEMENT_OPTIONS[0];
+    return result;
+  }
```

Comment:
Note: We cannot generalize these two methods into one, due to the large number of different possible interfaces for Query/Update/Read/QueryOrUpdate/QueryOrUpdateOrTransaction/Options.